### ISSUE TITLE
Fix S2234 FN: lambdas

### ISSUE THREAD

**Issue by antonioaversa** [SonarSource/sonar-dotnet#8072]
The current implementation of [S2234](https://sonarsource.github.io/rspec/#/rspec/S2234) doesn't take into lambdas,

```cs
var f = (int a, int b) => a + b;
var g = (int a, int b) => f(b, a); // FN
```

There are multiple scenarios in which false negatives are produced:
- lambda calling another lambda
- method calling a lambda
- local function calling a lambda
- partial inversion of arguments
- total inversion of arguments

Related issues:
- https://github.com/SonarSource/sonar-dotnet/issues/8070
- https://github.com/SonarSource/sonar-dotnet/issues/8071

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1683](https://sonarsource.atlassian.net/browse/NET-1683)

[NET-1683]: https://sonarsource.atlassian.net/browse/NET-1683?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
